File: flutter_inappwebview_ios/ios/Classes/Types/WebViewTransport.swift
//
//  WebViewTransport.swift
//  flutter_inappwebview
//
//  Created by Lorenzo Pichilli on 16/02/21.
//

import Foundation

public class WebViewTransport: NSObject {
    weak var webView: InAppWebView?
    var request: URLRequest
    
    init(webView: InAppWebView, request: URLRequest) {
        self.webView = webView
        self.request = request
    }
}

File: flutter_inappwebview_linux/linux/types/web_view_transport.cc
#include "web_view_transport.h"

#include "../in_app_webview/in_app_webview.h"

namespace flutter_inappwebview_plugin {

WebKitWebView* WebViewTransport::getWebKitWebView() const {
  if (inAppWebView) {
    return inAppWebView->webview();
  }
  return nullptr;
}

}  // namespace flutter_inappwebview_plugin
